src/compiler/factory.ts

```diff
@@ -291,33 +344,62 @@ namespace ts {
     }
 
     export function createPropertyAccess(expression: Expression, name: string | Identifier, location?: TextRange) {
-        const node = <PropertyAccessExpression>createNode(SyntaxKind.PropertyAccessExpression, location, /*flags*/ undefined);
+        return createPropertyAccessWithDotToken(expression, createSynthesizedNode(SyntaxKind.DotToken), name, location, /*flags*/ undefined);
+    }
+
+    export function createPropertyAccessWithDotToken(expression: Expression, dotToken: Node, name: string | Identifier, location?: TextRange, flags?: NodeFlags) {
+        const node = <PropertyAccessExpression>createNode(SyntaxKind.PropertyAccessExpression, location, flags);
         node.expression = parenthesizeForAccess(expression);
-        node.dotToken = createSynthesizedNode(SyntaxKind.DotToken);
+        node.dotToken = dotToken;
         node.name = typeof name === "string" ? createIdentifier(name) : name;
         return node;
     }
 
+    export function updatePropertyAccess(node: PropertyAccessExpression, expression: Expression, name: Identifier) {
+        if (node.expression !== expression || node.name !== name) {
+            return updateNode(createPropertyAccessWithDotToken(expression, node.dotToken, name, /*location*/ node, node.flags), node);
+        }
+        return node;
+    }
+
     export function createElementAccess(expression: Expression, index: number | Expression, location?: TextRange) {
         const node = <ElementAccessExpression>createNode(SyntaxKind.ElementAccessExpression, location);
         node.expression = parenthesizeForAccess(expression);
         node.argumentExpression = typeof index === "number" ? createLiteral(index) : index;
         return node;
     }
 
-    export function createCall(expression: Expression, argumentsArray: Expression[], location?: TextRange) {
-        const node = <CallExpression>createNode(SyntaxKind.CallExpression, location);
+    export function createCall(expression: Expression, typeArguments: TypeNode[], argumentsArray: Expression[], location?: TextRange, flags?: NodeFlags) {

```

Comment:
Why you need to pass in `typeArguments` parameter? In the change, it seems that all caller will pass `undefined`? also since we are in emitting phase, would we `typearguments` matter, won't we get rid of it anyway?